federation-js/src/composition/compose.ts

```diff
@@ -661,6 +662,7 @@ export function composeServices(services: ServiceDefinition[]): CompositionResul
     return {
       schema,
       composedSdl: printComposedSdl(schema, services),
+      coreSchema: printCoreSchema(schema),
```

Comment:
Do we need to continue to support both the old CSDL and core schema? If so, should we avoid printing both and expose those as separate print functions that act on the schema?

Reply:
I consider `composedSdl` to be something that we ultimately remove, but I think you make a fair point that it's excessive to do both prints. Seems like we have a few reasonable options:

0) leave as is and just remove it soon (and mark `composedSdl` as `@deprecated`?)
1) remove `composedSdl` in this PR as a breaking change 
2) `composeAndValidate` is just responsible for returning the `schema`, most consumers will then pass that to the `printXyz` fn of their choice - also breaking. Also, undeprecate `schema`. Not the most ergonomic end result, but maybe the most future proof?
3) ?

Reply:
Is anyone using `composedSdl`, or was that added just to support CSDL? What would break if we removed that today?

Reply:
Just returning a schema would only work if we were sure a `GraphQLSchema` could always fully represent a core schema. I'd love for that to be true, but currently there is no established mechanism for core schema metadata, and we just rely on our custom `extensions` for composition results. (I can see a world where we define a `CoreSchema` subclass that has further guarantees about how core schema metadata is represented under `extensions`,  but that's not something we can do in the short term.)

Reply:
Also, should `coreSchema` be `coreSchemaSDL` maybe?